### PR TITLE
git: replace gitToInit with lib function

### LINKED ISSUE #4354: programs.git: use `lib.generators.toGitINI` instead of `gitToIni`
Is there any good reason that `git.nix` defines an own function [`gitToIni`](https://github.com/nix-community/home-manager/blob/2471d965a3522025157a790fc49c3567fd56e26e/modules/programs/git.nix#L49)  instead of using [`lib.generators.toGitINI`](https://github.com/NixOS/nixpkgs/blob/cde2941c60c5bd68e99682af12745978ca4cd0d3/lib/generators.nix#L197C2-L197C2)?

If not, I'd like to normalize it by using the generator function.

### PR DESCRIPTION
**Pull request by VwCSXg**
### Description
Remove the function gitToInit and underlying functions in favor of using the function lib.generators.toGitINI from nixpkgs. 

closes #4354
depends on https://github.com/NixOS/nixpkgs/pull/250220

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef). 
  - works when run against https://github.com/VwCSXg/nixpkgs/tree/update-gitINI-generator: `nix develop --override-input nixpkgs 'git+https://github.com/VwCSXg/nixpkgs?ref=update-gitINI-generator' --ignore-environment .#all`

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

#### Maintainer CC
@rycee

### CODE REVIEW

File: tests/modules/programs/git/git-expected.conf [hunk 1]
```diff
@@ -1,64 +1,64 @@
...
-	path = "~/path/to/conditional.inc"
+	path = ~/path/to/conditional.inc
 
 [includeIf "gitdir:~/src/dir"]
-	path = "@git_include_path@"
+	path = /nix/store/q803rx1jj6pnfpza5dm8d8nkyi0aqwm4-hm_gitconfig
 
 [includeIf "gitdir:~/src/otherproject"]
-	path = "@git_named_include_path@"
+	path = /nix/store/y0c46dscy39gfhbd59b2vjjiysry5vpp-hm_gitconfigwork
```
Comment (ncfavier) [inline tests/modules/programs/git/git-expected.conf:64]: !

**Review by ncfavier — CHANGES_REQUESTED** [review]
The quote removal seems like a no-go: I have config values with `#` that definitely should be quoted. Why does the nixpkgs generator remove quotes?

**Comment by ncfavier** [comment]
Proper quoting was added in HM after the extraction: https://github.com/nix-community/home-manager/pull/1209

This will have to be upstreamed as well before this PR can proceed.

**Comment by VwCSXg** [comment]
> Proper quoting was added in HM after the extraction: #1209
Thanks for pointing that out, I missed that.

> This will have to be upstreamed as well before this PR can proceed.
You're right, I'll try that.

**Comment by rycee** [comment]
Thanks! Looks good now :-)

**Pull request merged**